jormungandr-lib/src/interfaces/vote/mod.rs

```diff
@@ -0,0 +1,69 @@
+use serde::Serialize;
+
+#[derive(Serialize)]
+#[serde(remote = "chain_impl_mockchain::block::BlockDate")]
+struct BlockDate {
+    pub epoch: u32,
+    pub slot_id: u32,
+}
+
+#[derive(Serialize)]
+#[serde(remote = "chain_impl_mockchain::certificate::ExternalProposalId")]
+struct ExternalProposalId {
+    #[serde(getter = "chain_impl_mockchain::certificate::ExternalProposalId::to_string")]
+    id: String,
+}
+
+#[derive(Serialize)]
+pub struct Proposal {
+    pub external_id: String,
+    pub options: u8,
```

Comment:
This is a bit more complex than just a `u8`. Can you make it an enum instead? So this can grow with time?

Right now, this byte represent a `bitmask` of the vote choices available, and only the first 4 bits are allowed. So if the option is `0b0011` the options are `0`, `1` or `2`. If the option is `0b0100`, the options are `0`, `1`, `2`, `3`.

Reply:
Could you make it an enum with only one entry?

```rust
enum VoteOption {
    OneOf { max_value: u8 } // where max_value is up to 15
}
```

Reply:
This will allow us to extend the model in the future without breaking APIs